driver/src/orderbook/batched_auction_data_reader.rs

```diff
@@ -0,0 +1,317 @@
+    ];
+    const ORDER_2_BYTES: &[u8] = &[
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, // user:
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 5, // sellTokenBalance: 5
+        1, 1, // buyToken: 256+1
+        1, 2, // sellToken: 256+2
+        0, 0, 0, 2, // validFrom: 2
+        0, 0, 1, 5, // validUntil: 256+5
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, // priceNumerator: 258;
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 3, // priceDenominator: 259
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, // remainingAmount: 2**8 = 256
+    ];
+    const ORDER_3_BYTES: &[u8] = &[
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, // user:
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 6, // sellTokenBalance: 6
+        1, 2, // buyToken: 256+2
+        1, 1, // sellToken: 256+1
+        0, 0, 0, 2, // validFrom: 2
+        0, 0, 1, 5, // validUntil: 256+5
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, // priceNumerator: 258;
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 3, // priceDenominator: 259
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, // remainingAmount: 2**8 = 256
+    ];
+
+    lazy_static! {
+        static ref ORDER_1: Order = Order {
+            id: 0,
+            account_id: H160::from_low_u64_be(1),
+            sell_token: 257,
+            buy_token: 258,
+            sell_amount: 257,
+            buy_amount: 257,
+        };
+        static ref ORDER_2: Order = Order {
+            id: 1,
+            account_id: H160::from_low_u64_be(1),
+            sell_token: 258,
+            buy_token: 257,
+            sell_amount: 256,
+            buy_amount: 256,
+        };
+    }
+
+    #[test]
+    fn batched_auction_data_reader_empty() {
+        let mut reader = BatchedAuctionDataReader::new(U256::from(3));
+        assert_eq!(reader.apply_batch(&[]), 0);
+    }
+
+    #[test]
+    fn batched_auction_data_reader_single_batch() {
+        let mut bytes = Vec::new();
+        bytes.extend(ORDER_1_BYTES);
+        bytes.extend(ORDER_2_BYTES);
+        let mut reader = BatchedAuctionDataReader::new(U256::from(3));
+        assert_eq!(reader.apply_batch(&bytes), 2);
+
+        let mut account_state = AccountState::default();
+        account_state.modify_balance(H160::from_low_u64_be(1), 257, |x| *x = 4);
+        account_state.modify_balance(H160::from_low_u64_be(1), 258, |x| *x = 5);
+
+        assert_eq!(reader.account_state, account_state);
+        assert_eq!(reader.orders, [ORDER_1.clone(), ORDER_2.clone()]);
+        assert_eq!(reader.pagination.previous_page_user, ORDER_1.account_id);
+        assert_eq!(reader.pagination.previous_page_user_offset, 2);
+    }
+
+    #[test]
+    fn batched_auction_data_reader_multiple_batches() {
```

Comment:
What are these Batches? Is this the same as auction index? Are these batches somehow related to "pages"?